### DISCUSSION TITLE
How to access locally defined phenology variables from a manager script

### DISCUSSION THREAD

**Discussion by pjinnes** [APSIMInitiative/ApsimX#9346] [Q&A] [answered]
I have created a 3 hourly interpolation called ThermalStress, similar to ThermalTime, in my RedClover phenology but can't find a way to access its value from a manager script. I can get the value in the Report Ok as [Phenology].ThermalStress.
From the script below I'm getting "Unable to cast object of type 'Models.Functions.SubDailyInterpolation' to type 'System.Double"
Any suggestions?

        [Link] private Clock clock;
        [Link] private IPlant crop;      
        [Link] private Summary Summary;
        [Link] private SimpleLeaf leaf;
        [Link] private BiomassRemoval biomassRemoval;
        [Link] private Phenology phenology;
        [Link] private Plant RedClover;
        [Link] private Zone zone;
                
        private double accumulatedTempStress = 0;
     
        [EventSubscribe("EndOfDay")]      
        private void OnDoManagementCalculations(object sender, EventArgs e)
        {       	
            if (phenology.CurrentPhaseName == "Reproductive")
            { 
            	double tempStress = (double)RedClover.FindByPath("Phenology.ThermalStress").Value;

![Screenshot 2024-09-25 121102](https://github.com/user-attachments/assets/691c3c47-0dce-459a-92b3-ad315fe62257)

**Comment by ric394** [comment]
The error message suggests that your `(double)` explicit cast is not correct. What type does the Report say `RedClover.FindByPath("Phenology.ThermalStress").Value` is?
The intellisense will let you know what type to cast to as below:
![image](https://github.com/user-attachments/assets/e29b14c5-4df8-4028-9533-57d49b572a7b)

**Reply by pjinnes** [reply]
thanks for that - helpful to know for next time

**Comment by hol353** [comment, marked as answer]
The easiest way to do this is to find the Function you want to call in the simulation tree (ThermalStress in your case), right click on it and select 'Copy manager snippet for model'. This will copy a snippet of script code to the clipboard that you can then paste into your manager script. The pasted code looks something like this:
```
using Models.Soils;
[Link(ByName=true)] private IFunction ThermalStress;
```
The snippet gives you the correct using statement which you should move to the using section of your script. The link statement should be moved to the link section of your manager script. When you want the value of ThermalStress you can just do:

```
double tempStress = ThermalStress.Value();
```

The manager tutorial on this page (https://apsimnextgeneration.netlify.app/modeldocumentation/) talks about this in more detail.

**Reply by pjinnes** [reply]
That works fine now - thanks for the info
